### ISSUE TITLE
Refactoring: more predictable filename handling

### ISSUE THREAD

**Issue by xrotwang** [lingpy/lingpy#215]
I just ran into the following problem: I specified an output file on the command line for the `pairwise` subcommand. To test the success of the command, I checked whether the specified file existed. This check failed, because `PSA.output` was called within the command, passing the filename as `filename` arg. But the output is actually written to a file with a different name:
```
outfile = filename + '.' + fileformat
```
This happens quite a bit in LingPy and - at least to me - is a continuous source for surprise and puzzlement.

At least when called from the command line, the exact filename specified should be used to write output to.

**Comment by LinguList** [comment]
Yes, I agree, I was also thinking about this when writing the first command line code yesterday, since now it's inconsequent: if you pass strings from the console, it will write to a file as one chooses, otherwise, if an input file is selected, it will use the method for file writing from PSA, MSA, or LexStat class.

One question for me would be: should we generally allow all kinds of filenames with endings to be written, or should we follow the matplotlib example (what I originally wanted to do), namely, determining output format by the specification of the filename with ending? For some reason which seems less clear now, I was convinced at some point that the splitting output("format", filename='xxx') was a good idea. Well, one keeps learning. Yet, I still think that the matplotlib way to treat file formats is actually nice: output("wordlist.tsv") vs. output("wordlist.html") would just trigger the different formats.

**Comment by xrotwang** [comment]
I think inferring a course of action from the filename extension is ok. That's what most people are used to from their OS after all. But I would limit this cleverness to the case where no `format` argument is passed. So in general, I'd say, functions should accept a `filename` argument and stick to this name, when writing output. No automagic appended extension, no silent name manipulation if a file with the given name already exists.

Regarding matplotlib: AFAICT this library is always mentioned as an example for confusing user interface  or API design. This weird global `rcParams`, for example, is something I'd like to see replaced in LingPy as well.

**Comment by LinguList** [comment]
It seems matplotlib was a bad "teacher" I had ;-)

But regarding parameters, how would you handle global things, like gap penalties, etc., since I think that it definitely makes sense to have them defined rather ones than always again (already now, the defaults vary to some degree, e.g., in pairwise and mutliple alignments, and having a global storage for parameters seems like a good answer to me, although I gladly replace rcParams by something more consistent...).

**Comment by xrotwang** [comment]
I think that's what config files are for. Config files (preferably in INI format) have some advantages:
- There is a mostly clear policy for each OS where to put custom config files.
- Users - even ones with no knowledge of python - can edit them.

I'd even count the limited expressiveness as opposed to python code as advantage. 

Right now, it is really difficult to be sure the global `rcParams` are not changed somewhere within an analysis.
While this could also happen with a global config object, initialized from a file, this will at least not happen by accident, e.g. by importing modules in unexpected order.

**Comment by LinguList** [comment]
I see. These config files could then potentially also handle all the rc-files in data/conf/, I guess?

Maybe, as a first homework for myself, I should try to find time and settle all the major keywords that I deem indispensable to be on the ini-basis, like gap-opening-penalty, etc., and just write them down in a first INI file...

**Comment by xrotwang** [comment]
You said that for different subcommands different values for the same options may make sense. So maybe we should have one section per command in the config file, with a global [`DEFAULT` section](https://docs.python.org/3/library/configparser.html#quick-start). So the INI file could look like:
```
[DEFAULT]
gop = -1.0

[cmd_alignments]
gop = -0.9
```

**Comment by LinguList** [comment]
yes, that is a good idea!
